hibernate-core/src/main/java/org/hibernate/persister/entity/EntityPersister.java

```diff
@@ -1129,6 +1130,16 @@ default Object getValue(Object object, int i) {
 	 */
 	Object getIdentifier(Object entity, SharedSessionContractImplementor session);
 
+	/**
+	 * Get the identifier of an instance from the object's identifier property.
+	 * Throw an exception if it has no identifier property.
+	 *
+	 * It's supposed to be use during the merging process
+	 */
+	default Object getIdentifier(Object entity, MergeContext mergeContext, SharedSessionContractImplementor session) {
```

Comment:
It seems to me that access to the `session` could be gained through the `MergeContext`, so passing the session separately is unnecessary.

Reply:
Thanks,@beikov,
change made